src/unk_0203E880.c

```diff
@@ -1,596 +1,476 @@
+    } else if (retScriptID >= 9800) {
+        ScriptContext_Load(fieldSystem, ctx, 212, 217);
+        retScriptID -= 9800;
+    } else if (retScriptID >= 9700) {
+        ScriptContext_Load(fieldSystem, ctx, 422, 429);
+        retScriptID -= 9700;
+    } else if (retScriptID >= 9600) {
+        ScriptContext_Load(fieldSystem, ctx, 412, 213);
+        retScriptID -= 9600;
+    } else if (retScriptID >= 9500) {
+        ScriptContext_Load(fieldSystem, ctx, 501, 547);
+        retScriptID -= 9500;
+    } else if (retScriptID >= 9400) {
+        ScriptContext_Load(fieldSystem, ctx, 426, 432);
+        retScriptID -= 9400;
+    } else if (retScriptID >= 9300) {
+        ScriptContext_Load(fieldSystem, ctx, 406, 374);
+        retScriptID -= 9300;
+    } else if (retScriptID >= 9200) {
+        ScriptContext_Load(fieldSystem, ctx, 423, 430);
+        retScriptID -= 9200;
+    } else if (retScriptID >= 9100) {
+        ScriptContext_Load(fieldSystem, ctx, 0, 11);
+        retScriptID -= 9100;
+    } else if (retScriptID >= 9000) {
+        ScriptContext_Load(fieldSystem, ctx, 213, 221);
+        retScriptID -= 9000;
+    } else if (retScriptID >= 8970) {
+        ScriptContext_Load(fieldSystem, ctx, 425, 7);
+        retScriptID -= 8970;
+    } else if (retScriptID >= 8950) {
+        ScriptContext_Load(fieldSystem, ctx, 498, 539);
+        retScriptID -= 8950;
+    } else if (retScriptID >= 8900) {
+        ScriptContext_Load(fieldSystem, ctx, 424, 431);
+        retScriptID -= 8900;
+    } else if (retScriptID >= SCRIPT_ID_OFFSET_SAFARI_ZONE) {
+        ScriptContext_Load(fieldSystem, ctx, 497, 538);
+        retScriptID -= SCRIPT_ID_OFFSET_SAFARI_ZONE;
+    } else if (retScriptID >= SCRIPT_ID_OFFSET_HIDDEN_ITEMS) {
+        ScriptContext_Load(fieldSystem, ctx, 408, 380);
+        retScriptID -= SCRIPT_ID_OFFSET_HIDDEN_ITEMS;
+    } else if (retScriptID >= 7000) {
+        ScriptContext_Load(fieldSystem, ctx, 404, 369);
+        retScriptID -= 7000;
+    } else if (retScriptID >= SCRIPT_ID_OFFSET_DOUBLE_BATTLES) {
+        ScriptContext_Load(fieldSystem, ctx, 1114, 213);
+        retScriptID -= SCRIPT_ID_OFFSET_DOUBLE_BATTLES;
+    } else if (retScriptID >= SCRIPT_ID_OFFSET_SINGLE_BATTLES) {
+        ScriptContext_Load(fieldSystem, ctx, 1114, 213);
+        retScriptID -= SCRIPT_ID_OFFSET_SINGLE_BATTLES;
+    } else if (retScriptID >= 2800) {
+        ScriptContext_Load(fieldSystem, ctx, 413, 397);
+        retScriptID -= 2800;
+    } else if (retScriptID >= 2500) {
+        ScriptContext_Load(fieldSystem, ctx, 1, 17);
+        retScriptID -= 2500;
+    } else if (retScriptID >= 2000) {
+        ScriptContext_Load(fieldSystem, ctx, 211, 213);
+        retScriptID -= 2000;
+    } else if (retScriptID >= 1) {
+        ScriptContext_SetMapScripts(fieldSystem, ctx);
+        retScriptID -= 1;
     } else {
-        sub_0203EF14(fieldSystem, param1, 402, 355);
-        v0 = 0;
+        ScriptContext_Load(fieldSystem, ctx, 402, 355);
```

Comment:
**question:** Do we have constants for these bank IDs? What about the script NARC indices?

Reply:
I don't think we have constants for either of these yet. That's something I'm looking to do soon, fill out more of the scripting-related constants in general.